src/view/DashboardView.java

```diff
@@ -0,0 +1,239 @@
+import java.util.concurrent.ScheduledFuture;
+import java.util.concurrent.TimeUnit;
+import javax.swing.text.MaskFormatter;
+
+public class DashboardView extends JFrame implements PropertyChangeListener {
+    private final DashboardViewModel dashboardViewModel;
+    private JPanel dashboardPanel;
+    private ArrayList<ProjectData> projectsList;
+    private final ScheduledExecutorService schedule = Executors.newScheduledThreadPool(1);
+    private ScheduledFuture<?> scheduledFuture = null;
+
+    public DashboardView(DashboardViewModel dashboardViewModel, AddProjectController addProjectController,
+                         NotificationController notificationController) {
+        this.dashboardViewModel = dashboardViewModel;
+
+        DashboardState dashboardState = dashboardViewModel.getState();
+
+        dashboardViewModel.addPropertyChangeListener(this);
+
+        projectsList = dashboardState.getProjects();
+
+        displayAllProjects();
+
+        setTitle("Project Dashboard");
+        setSize(600, 400);
+        setDefaultCloseOperation(JFrame.EXIT_ON_CLOSE);
+        setLayout(new BorderLayout());
+        setResizable(false);
+
+        dashboardPanel = new JPanel();
+        dashboardPanel.setLayout(new BoxLayout(dashboardPanel, BoxLayout.Y_AXIS));
+
+        JScrollPane scrollPane = new JScrollPane(dashboardPanel);
+        scrollPane.setVerticalScrollBarPolicy(ScrollPaneConstants.VERTICAL_SCROLLBAR_ALWAYS);
+
+        add(scrollPane, BorderLayout.CENTER);
+
+        updateEmptyDashboardLabel();
+
+        JButton addProjectButton = new JButton("Add Project");
+        JButton toggleNotifications = new JButton("Notifications Off");
+        addProjectButton.addActionListener(new ActionListener() {
+            @Override
+            public void actionPerformed(ActionEvent e) {
+                if (e.getSource() == addProjectButton) {
+                    try {
+                        showAddProjectPopup();
+                    } catch (ParseException ex) {
+                        throw new RuntimeException(ex);
+                    }
+                }
+            }
+        });
+        toggleNotifications.addActionListener(new ActionListener() {
+            @Override
+            public void actionPerformed(ActionEvent e) {
+                if (e.getSource() == toggleNotifications) {
+                    if (scheduledFuture == null) {
+                        Runnable sendNotification = () -> notificationController.execute(LocalDate.now());
+                        scheduledFuture = schedule.scheduleAtFixedRate(sendNotification, 0, 24, TimeUnit.HOURS);
+                        toggleNotifications.setText("Notifications Off");
+                    }
+                    else {
+                        scheduledFuture.cancel(true);
+                        scheduledFuture = null;
+                        toggleNotifications.setText("Notifications On");
+                    }
```

Comment:
some comments explaining the action when clicking the toggleNotification button would be convinient.